src/scripts/annotate.pl

```diff
@@ -246,37 +246,51 @@
 }
 
 my %bad_fusions;
-if(!$internal){
+#if(!$internal){
 	my $df = new DelimitedFile(
 	       "-file" => $blacklist_fusion_file,
 	       "-headers" => 1,
 	      );
 
 	while (my $row = $df->get_hash()) {
+	      if($row->{geneA} eq '') {$row->{geneA} = "NA";}
+	      if($row->{geneB} eq '') {$row->{geneB} = "NA";}
+
 	      my @gene_a = split(",", $row->{geneA});
 	      my @gene_b = split(",", $row->{geneB});
 	      my ($chrA, $posA, $chrB, $posB) = ($row->{chrA}, $row->{posA}, $row->{chrB}, $row->{posB});
+		#print STDERR "badfusions ".$chrA." ".$posA." ".$chrB." ".$posB."\n";
 	      foreach my $g1 (@gene_a){
-			$g1 = join(":", $chrA, $posA) if($g1 eq 'NA');
-		foreach my $g2 (@gene_b){
+		   next if($g1 eq ''); 
+		   $g1 = join(":", $chrA, $posA) if($g1 eq 'NA');
+		   
+		   foreach my $g2 (@gene_b){
+             		next if($g2 eq '');
 			$g2 = join(":", $chrB, $posB) if($g2 eq 'NA');
 			my $gene_pair = ($g1 le $g2) ? join(":",$g1,$g2) : join(":",$g2,$g1);
-			$bad_fusions{$gene_pair} = [$chrA, $posA, $chrB, $posB];
-			}
+			$bad_fusions{$chrA.":".$posA.":".$chrB.":".$posB} = $gene_pair;
+			#$bad_fusions{$gene_pair} = [$chrA, $posA, $chrB, $posB];
```

Comment:
Please delete the code that is no longer used, rather than just commenting it out.